### PR TITLE
add priorityclass support for node exporter and fluentd

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

Add priority class support for both node exporter and fluentd daemonset to make it schedulable and not get blocked due to resource crunch and allow more priority

## Related Issues

https://github.com/astronomer/issues/issues/6449

## Testing

Yet to update

## Merging

Yet to update

### CODE REVIEW

**Review by danielhoherd — CHANGES_REQUESTED** [review]
Needs tests.

File: tests/chart_tests/test_fluentd.py [hunk 1]
```diff
@@ -300,3 +307,31 @@ def test_fluentd_index_overrides(self, kube_version):
...
+            values={},
+            show_only=["charts/fluentd/templates/fluentd-daemonset.yaml"],
+        )
+        assert len(docs) == 1
+        doc = docs[0]
+        self.common_tests_daemonset(doc)
+        assert "priorityClassName" not in doc["spec"]["template"]["spec"]
+
+    def test_fluentd_priorityclass_overrides(self, kube_version):
+        """Test to validate fluentd with priority class ."""
```
Comment (danielhoherd) [inline tests/chart_tests/test_fluentd.py:324]:
```suggestion
        """Test to validate fluentd with priority class configured."""
```

File: tests/chart_tests/test_fluentd.py [hunk 2]
```diff
@@ -300,3 +307,31 @@ def test_fluentd_index_overrides(self, kube_version):
             "index_patterns": ["astronomer.*"],
             "mappings": {"properties": {"date_nano": {"type": "date_nanos"}}},
         }
+
+    def test_fluentd_priorityclass_defaults(self, kube_version):
+        """Test to validate fluentd with priority class ."""
```
Comment (danielhoherd) [inline tests/chart_tests/test_fluentd.py:312]:
```suggestion
        """Test to validate fluentd with priority class defaults."""
```

File: tests/chart_tests/test_fluentd.py [hunk 3]
```diff
@@ -10,6 +10,13 @@
     supported_k8s_versions,
 )
 class TestFluentd:
+
+    @staticmethod
+    def common_tests_daemonset(doc):
+        """Test things common to all daemonsets."""
+        assert doc["kind"] == "DaemonSet"
+        assert doc["metadata"]["name"] == "release-name-fluentd"
+
```
Comment (danielhoherd) [inline tests/chart_tests/test_fluentd.py:19]: Unrelated to the changes so far, the docstring on lines 21-22 needs to be updated.

File: tests/chart_tests/test_prometheus_node_exporter.py [hunk 4]
```diff
@@ -107,3 +111,35 @@ def test_prometheus_node_exporter_daemonset_with_security_context_overrides(
             "readOnlyRootFilesystem": True,
             "runAsNonRoot": True,
         }
+
+    def test_node_exporter_priorityclass_defaults(self, kube_version):
+        """Test to validate fluentd with priority class ."""
```
Comment (danielhoherd) [inline tests/chart_tests/test_prometheus_node_exporter.py:116]:
```suggestion
        """Test to validate fluentd with priority class defaults."""
```

File: tests/chart_tests/test_prometheus_node_exporter.py [hunk 5]
```diff
@@ -107,3 +111,35 @@ def test_prometheus_node_exporter_daemonset_with_security_context_overrides(
...
+            values={},
+            show_only=["charts/prometheus-node-exporter/templates/daemonset.yaml"],
+        )
+        assert len(docs) == 1
+        doc = docs[0]
+        self.common_tests_daemonset(doc)
+        assert "priorityClassName" not in doc["spec"]["template"]["spec"]
+
+    def test_node_exporter_priorityclass_overrides(self, kube_version):
+        """Test to validate node exporter with priority class ."""
```
Comment (danielhoherd) [inline tests/chart_tests/test_prometheus_node_exporter.py:128]:
```suggestion
        """Test to validate node exporter with priority class configured."""
```

**Review by danielhoherd — APPROVED** [review]
Code looks good! I left a few docstring suggestions.

File: tests/chart_tests/test_fluentd.py (hunk 3 above)
Comment (pgvishnuram) [reply tests/chart_tests/test_fluentd.py:19]: i created this function to call them verify the kind and apiVersion

**Comment by pgvishnuram** [comment]
Addressed all the requested changes merge them

**Pull request merged**